install/gcp/deployment_manager/istio-cluster.jinja

```diff
@@ -129,12 +138,65 @@ resources:
           helm template install/kubernetes/helm/istio --name istio --namespace istio-system $ISTIO_OPTIONS  > istio.yaml
 
           kubectl apply -f istio.yaml
-          kubectl label namespace default istio-injection=enabled
 
-          sleep 150
+          {% if  properties['enableAutomaticSidecarInjection'] %}
+            kubectl label namespace default istio-injection=enabled
+          {% endif %}
+
+          until kubectl rollout status deployment.extensions/istio-pilot -n istio-system ; do sleep 1 ; done
+          until kubectl rollout status deployment.extensions/istio-citadel -n istio-system ; do sleep 1 ; done
+          until kubectl rollout status deployment.extensions/istio-policy -n istio-system ; do sleep 1 ; done
+          until kubectl rollout status deployment.extensions/istio-galley -n istio-system ; do sleep 1 ; done
+          until kubectl rollout status deployment.extensions/istio-ingressgateway -n istio-system ; do sleep 1 ; done
+
+          {% if  properties['enableKiali'] %}
+
+            export USERNAME=$(echo -n 'admin' | base64)
```

Comment:
This isn't correct.  setting Kiali username and passwords to defaults is a sure way to problems.  This was fixed in master this way:
https://preliminary.istio.io/docs/tasks/telemetry/kiali/#before-you-begin

(IOW ask for a password and then enter it in).

I am not sure if this template runs interactively.  If it does, the above docs should help you get there.

Alternatively, you could turn off Kiali, although I personally feel it adds alot of value.

Reply:
The template is interactive in the sense it can get userinput (e,g username/password) and inject it into the property set within the startup script.  However, i don't know how to construct the template such that is conditonal that it not optional once someone selects 'enableKiali`.  There maybe a way but i don't have time to look into that.

--

re: default credentials. that is an issue but the mode of access provided here is to use `kube-proxy` to get to the console.  This is similar to `prometheus`, `servicegraph`, `grafana`, etc which doens't even have auth built in.  GIven that, seems ok to leave it as is.  If we do have it truly external (i.,e no kube-proxy), then we can PR a separate change in for an option to get userinput.